src/main/java/org/aion/wallet/ui/components/HistoryController.java

```diff
@@ -127,6 +151,28 @@ private void setEventHandlers() {
         copyItem.setOnAction(new ContextMenuTableCopyEventHandler(txTable));
         menu.getItems().add(copyItem);
         txTable.setContextMenu(menu);
+
+
+        searchField.textProperty().addListener((observable, oldValue, newValue) -> {
+            FilteredList<TxRow> filteredData = new FilteredList(FXCollections.observableList(completeTransactionList), s -> true);
+            if (!newValue.isEmpty()) {
+                filteredData.setPredicate(s -> anyFieldHasString(s, newValue));
+                txTable.setItems(filteredData);
+            } else {
+                filteredData.setPredicate(s -> true);
+            }
+        });
+    }
+
+    private boolean anyFieldHasString(TxRow s, String newValue) {
```

Comment:
`final` params?
since you're matching strings only with selected `searchItem` maybe this could have a better name